aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -928,6 +1049,36 @@ void convert_indices_from_csr_to_coo_cpu(
         }
       });
 }
+template <typename input_t, typename output_t>
+void compressed_to_batched_compressed_indices_cpu(
+    const Tensor& batched_out,
+    const Tensor& compressed_in,
+    const int64_t n_batch) {
```

Comment:
Could you please clarify what this function is doing? We can't do `expand + contiguous`?

Reply:
Given the compressed indices for a matrix with compressed axis of length `l` we return the compressed indices for a batched - sparse tensor with batch length `n_batch` and compressed axis length `l / n_batch`.

The shapes are not quite compatible for an easy path using high-level functions `compressed_in` being `(nnz + 1, )` and `batched_out` being `(n_batch, nnz_per_batch +1)`

This function would also be useful if we were to implement a reshape operation that allowed the user to factor the compressed axis into a new batch dimension. Which is almost what we are doing in the dense->sparse-compressed conversions. There would need to be different checks upfront to make sure the sparsity allows that re-shape though.

Reply:
@nikitaved Let me know if the comments help.